controlplane/kubeadm/controllers/kubeadm_control_plane_controller_test.go

```diff
@@ -1357,234 +1344,34 @@ func TestKubeadmControlPlaneReconciler_scaleUpControlPlane(t *testing.T) {
-				controlPlaneUnHealthy: true,
-				selectedAnnotationKeys: []string{
-					controlplanev1.DeleteForScaleDownAnnotation,
-					controlplanev1.ScaleDownEtcdMemberRemovedAnnotation,
-				},
-			},
-			{
-				name:                  "both health check fails",
-				etcdUnHealthy:         true,
-				controlPlaneUnHealthy: true,
-				selectedAnnotationKeys: []string{
-					controlplanev1.DeleteForScaleDownAnnotation,
-				},
-			},
-		}
-		for _, tc := range testCases {
-			g := NewWithT(t)
-
-			fakeClient := newFakeClient(g, initObjs...)
-			fmc := &fakeManagementCluster{
-				Machines:            beforeMachines.DeepCopy(),
-				ControlPlaneHealthy: !tc.controlPlaneUnHealthy,
-				EtcdHealthy:         !tc.etcdUnHealthy,
-			}
-
-			r := &KubeadmControlPlaneReconciler{
-				Client:            fakeClient,
-				managementCluster: fmc,
-				Log:               log.Log,
-				recorder:          record.NewFakeRecorder(32),
-			}
-
-			ownedMachines := fmc.Machines.DeepCopy()
-			_, err := r.scaleDownControlPlane(context.Background(), cluster.DeepCopy(), kcp.DeepCopy(), ownedMachines)
-			g.Expect(err).To(HaveOccurred())
-			g.Expect(err).To(MatchError(&capierrors.RequeueAfterError{RequeueAfter: HealthCheckFailedRequeueAfter}))
-
-			controlPlaneMachines := &clusterv1.MachineList{}
-			g.Expect(fakeClient.List(context.Background(), controlPlaneMachines)).To(Succeed())
-			g.Expect(controlPlaneMachines.Items).To(HaveLen(len(beforeMachines)))
-
-			// We expect that a machine has been marked for deletion, since that isn't blocked by health checks
-			endMachines := internal.NewFilterableMachineCollectionFromMachineList(controlPlaneMachines)
-			selected := endMachines.Filter(internal.HasAnnotationKey(controlplanev1.DeleteForScaleDownAnnotation))
-			g.Expect(selected).To(HaveLen(1))
-			for _, m := range selected {
-				cm := m.DeepCopy()
-
-				g.Expect(m.Annotations).To(HaveLen(len(tc.selectedAnnotationKeys)))
-				for _, key := range tc.selectedAnnotationKeys {
-					g.Expect(m.Annotations).To(HaveKey(key))
-				}
-
-				// Remove the annotations and resource version to compare against the before copy
-				cm.Annotations = nil
-				cm.ResourceVersion = ""
-				bm, ok := beforeMachines[cm.Name]
-				g.Expect(ok).To(BeTrue())
-				g.Expect(cm).To(Equal(bm))
-			}
+			ControlPlaneHealthy: true,
+		},
+	}
 
-			// Ensure the non-selected machine match the before machines exactly
-			notSelected := endMachines.Filter(internal.Not(internal.HasAnnotationKey(controlplanev1.DeleteForScaleDownAnnotation)))
-			for _, m := range notSelected {
-				bm, ok := beforeMachines[m.Name]
-				g.Expect(ok).To(BeTrue())
-				g.Expect(m).To(Equal(bm))
-			}
-		}
-	})
+	_, err := r.scaleDownControlPlane(context.Background(), &clusterv1.Cluster{}, &controlplanev1.KubeadmControlPlane{}, machines)
+	g.Expect(err).ToNot(HaveOccurred())
 }
 
-func TestKubeadmControlPlaneReconciler_upgradeControlPlane(t *testing.T) {
```

Comment:
Are these tests not useful anymore, should they be moved somewhere else or actually deleted?

Reply:
aka, is this the test you mentioned in the PR description as "large pseudo integration test"?

Reply:
yes, this one and scale down. i know @dlipovetsky, who implemented scale down, was also unhappy with the tests and is working to write an e2e test for that instead.